runtime/jvmti/jvmtiHook.c

```diff
@@ -2139,8 +2150,9 @@ jvmtiHookPopFramesInterrupt(J9HookInterface** hook, UDATA eventNum, void* eventD
 static void
 jvmtiHookResourceExhausted(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
 {
-	J9JVMTIEnv * j9env = userData;
+	J9JVMTIEnv *j9env = userData;
 	jvmtiEventResourceExhausted callback = j9env->callbacks.ResourceExhausted;
+	J9VMThread *currentThread = ((J9VMVmResourceExhaustedEvent *)eventData)->currentThread;
```

Comment:
Earlier, I had missed that `data` was also being used on lines 2171 and 2174. We will need a local variable. @thallium Your initial impl was correct.
```suggestion
	J9VMVmResourceExhaustedEvent * data = (J9VMVmResourceExhaustedEvent *) eventData;
	J9VMThread * currentThread = data->currentThread;
```